contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -983,10 +983,10 @@ AND has_table_privilege(t1.oid, 'SELECT,INSERT,UPDATE,DELETE,TRUNCATE,TRIGGER');
 GRANT SELECT ON sys.sp_tables_view TO PUBLIC;
 
 CREATE OR REPLACE FUNCTION sys.sp_tables_internal(
-	in_table_name sys.nvarchar(384) = '',
-	in_table_owner sys.nvarchar(384) = '', 
-	in_table_qualifier sys.sysname = '',
-	in_table_type sys.varchar(100) = '',
+	in_table_name sys.nvarchar(384) = NULL,
```

Comment:
Why are we changing '' to NULL? NULL and '' are not same, so on what basis are we doing this?  What is the output of tsql if we EXEC the proc sp_tables? Does it return NULL or empty rows?

Reply:
Our current implementation treats both NULL and `''` similarly during the lookup as shown below:
```
WHERE ((SELECT coalesce(in_table_name,'')) = '' OR table_name = in_table_name collate sys.database_default)
AND ((SELECT coalesce(in_table_owner,'')) = '' OR table_owner = in_table_owner collate sys.database_default)
AND ((SELECT coalesce(in_table_qualifier,'')) = '' OR table_qualifier = in_table_qualifier collate sys.database_default)
AND ((SELECT coalesce(cs_as_in_table_type,'')) = ''
```

Also, as per T-SQL, default parameter values are `NULL` not `''`, but because of our current implementation, we were getting output differences w.r.t T-SQL (Added test cases for the same).

I hope this clears the motive behind changing the default parameter values.